tests/models/test_signature.py

```diff
@@ -130,6 +130,27 @@ def test_signature_inference_infers_input_and_output_as_expected():
     assert sig1.outputs == sig0.inputs
 
 
+def test_infer_signature_on_list_of_dictionaries():
+    signature = infer_signature(
+        model_input=[{"query": "test query"}],
+        model_output=[
+            {
+                "output": "Output from the LLM",
+                "candidate_ids": ["412", "1233"],
```

Comment:
Should the type of a list of strings be a string?

Reply:
Yea that's also my question.. but https://github.com/mlflow/mlflow/blob/f9aae45bc5ebc900d08b27f00ac835e3cb8961e7/mlflow/types/utils.py#L143-L145 also just uses ColSpec

Reply:
I think this is one of the limitations. `DataType` is not expressive enough to support compound types.

Reply:
Or should we use TensorSpec instead?

Reply:
Can TensorSpec be named?

Reply:
Yes def __init__(
        self,
        type: np.dtype,  # pylint: disable=redefined-builtin
        shape: Union[tuple, list],
        name: Optional[str] = None,
    ):

Reply:
`TensorSpec` seems more reasonable and less confusing (to me).

Reply:
Yes, I agree, then we should change together as https://github.com/mlflow/mlflow/blob/f9aae45bc5ebc900d08b27f00ac835e3cb8961e7/mlflow/types/utils.py#L143-L145 to be consistent. WDYT? @BenWilson2

Reply:
For consistency and future work on standardizing the output format for pyfunc return values, I feel like sticking with ColSpec will make the most sense here (particularly with handling pandas_udf and spark_udf integrations).

Reply:
+1 to @BenWilson2 's comment here. Let's try to stick with ColSpec since it's friendlier for tabular data